compiler/closureiters.nim

```diff
@@ -606,6 +606,13 @@ proc lowerStmtListExprs(ctx: var Ctx, n: PNode, needsSplit: var bool): PNode =
             internalError(ctx.g.config, "lowerStmtListExpr(nkCaseStmt): " & $branch.kind)
         result.add(n)
         result.add(ctx.newEnvVarAccess(tmp))
+      else:
+        result = newNodeI(nkStmtList, n.info)
+        if n[0].kind == nkStmtListExpr:
```

Comment:
what if it's not a stmtlistexpr?

Reply:
Then it cannot contain a `yield`. I think. :-)

Reply:
shouldn't it.. error out then? just in case?

Reply:
Well previously the case wasn't handled and it produced an internal error. If this assumption doesn't hold, it will produce the internal error again:

```nim

  if n.hasYieldsInExpressions():
    internalError(ctx.g.config, "yield in expr not lowered")

```

Reply:
oh. hm, that's a really really hard message to debug, it should at least include the offending node

Reply:
We can improve that in a follow-up PR.